### PR TITLE
Actually include attribute-defined properties in patch computation

### LINKED ISSUE #929: Updates for Ref properties specified with attributes are always `null` in `/api/subscribe` responses
To reproduce:
1. Run `rojo plugin install`
2. Run `rojo serve` on the [`ref_properties` test project](https://github.com/rojo-rbx/rojo/tree/7e2bab921aa71f76d07d0424e4bb4064e8b7c995/rojo-test/serve-tests/ref_properties)
3. Open a fresh baseplate in Roblox Studio, enable trace logging in the Rojo plugin, and click Connect
4. Make any change to ref_properties' `default.project.json` (I added a newline)
5. Observe the output, and notice that the subscribe response contains `null` for all the `Ref` properties specified with attributes in ref_properties' `default.project.json`:
```json
{
  "sessionId": "d264957a-76ce-486b-8df6-7151de8da0c1",
  "messageCursor": 2,
  "messages": [
    {
      "removed": [],
      "added": {},
      "updated": [
        {
          "id": "516ae8375fd349b4a61ac47529c969cf",
          "changedName": null,
          "changedClassName": null,
          "changedProperties": {
            "Value": null
          },
          "changedMetadata": null
        },
        {
          "id": "1aee8e1cc88e674f9917cc95e9d56bf6",
          "changedName": null,
          "changedClassName": null,
          "changedProperties": {
            "Value": null
          },
          "changedMetadata": null
        },
        {
          "id": "5e3e89750a0826e8d5d83b1ee86cad1a",
          "changedName": null,
          "changedClassName": null,
          "changedProperties": {
            "PrimaryPart": null
          },
          "changedMetadata": null
        },
        {
          "id": "04b43e42134d58c42408ff2b9ba1c9a4",
          "changedName": null,
          "changedClassName": null,
          "changedProperties": {
            "Value": null
          },
          "changedMetadata": null
        }
      ]
    }
  ]
}
``` 
I noticed this behavior in some test failures after #843 was merged: https://github.com/rojo-rbx/rojo/actions/runs/9605402645/job/26492956529#step:7:762. This makes it impossible for the Rojo plugin to live update a Ref property, could cause problems once rbx_dom_lua gains better ability to represent optional values, and can currently cause test failures on macOS due to seemingly inherent noisiness in FSEvents. 

One reason this could happen is because deferred Ref property application appears to not influence the returned result of patch application (i.e. `PatchApplyContext.applied_patch_set`), but rather only influences the tree:
https://github.com/rojo-rbx/rojo/blob/7e2bab921aa71f76d07d0424e4bb4064e8b7c995/src/snapshot/patch_apply.rs#L246-L299

It will also be worth investigating patch computation, to see if Ref properties specified via attributes are erroneously detected as removed there.

### PR DESCRIPTION
**Pull request by Dekkonot**
Closes #929.

It turns out the underlying issue was that we weren't accounting for ref properties at all in the patch computation. This resolves that issue by just collecting them during `compute_property_patch`.

Of note, I've decided to just reuse the `ref_properties` test files for this since it's ultimately academic either way. It adds noise to the snapshot files but I think it's fine because the only thing that really matters here is the `messageCursor` on the return from the read endpoint. Let me know if you'd prefer a cleaner snapshot.

### CODE REVIEW

**Comment by Dekkonot** [comment]
And of course, like clockwork, tests start failing on MacOS.

I promise they all pass on Windows. :-)

**Comment by kennethloeffler** [comment]
Code looks good, and tests pass for me on macOS, at least most of the time...

These test failures have been occurring because macOS' FSEvents API occasionally reports events that occurred *before* the serve session starts - i.e. in here https://github.com/rojo-rbx/rojo/blob/3ca975d81dfd6f06e024dec5a87bedf5f796dfc2/tests/rojo_test/serve_util.rs#L77-L98 
we're sometimes getting events from copying the test files to temporary directories. The only public information I can find about this behavior is this Stack Overflow question: https://stackoverflow.com/questions/47679298/howto-avoid-receiving-old-events-in-fseventstream-callback-fsevents-framework-o. It seems to happen more often when the system is under higher load, but I haven't been able to make any hard correlations.

I haven't been able to find any way we can avoid this. At this point, I'm thinking we should just sleep in between copying files and starting the server, hope it's enough time for FSEvents to not report these events, and be done with it. I've done this here: #945

**Comment by Dekkonot** [comment]
Good news: that seems to have done it. Hopefully we can find a solution that isn't that at some point, but for now I think MacOS tests taking slightly longer isn't that bad.

File: tests/tests/serve.rs [hunk 1]
```diff
@@ -477,3 +477,36 @@ fn ref_properties_remove() {
         );
     });
 }
+
+#[test]
+fn ref_properties_patch_update() {
```
Comment (kennethloeffler) [inline tests/tests/serve.rs:482]: Maybe add some comments to this test to describe what we expect to happen here, and why this test was added? Maybe link to the issue

**Comment by Dekkonot** [comment]
Comment: added.

**Review by kennethloeffler — APPROVED** [review]

**Pull request merged**